### DISCUSSION TITLE
[firebase_remote_config] Fetch is slow on first startup

### DISCUSSION THREAD

**Discussion by dev-tatsuya** [firebase/flutterfire#12234] [Q&A]
There is an app that uses RemoteConfig to A/B test the screen that is displayed when the app is launched for the first time.
When I installed this app on a device (emulator) that had never been installed, it took about 1 minute and 30 seconds to start.
When I debugged it, I found that it took about 1 minute and 30 seconds to fetch the RemoteConfig.
Indeed, [this article](https://firebase.google.com/docs/remote-config/loading#:~:text=dramatic%20visual%20changes.-,Note%3A%20If%20you%20are%20loading%20values%20for%20an%20A/B%20Testing%20experiment%2C%20this%20strategy%20is%20very%20strongly%20recommended.%20With%20A/B%20Testing%2C%20additional%20time%20is%20required%20as%20the%20user%20is%20placed%20into%20an%20experiment%20and%20the%20experimental%20values%20are%20applied.,-Strategy%203%3A%20Load) states that fetching RemoteConfig for A/B testing takes time.
However, I think an app that takes 1 minute and 30 seconds to start up for the first time is way too long. We are concerned about user withdrawal.
So, as posted, how can I use RemoteConfig to A/B test the screen that is displayed at first startup so that the app opens quickly on first startup?
The code flow is to await the RemoteConfig `fetch` before `runApp` and pass the obtained value to the Widget inside `runApp`.

The version is below.
```
  firebase_remote_config: ^4.2.4
```

Thank you for your help.

**Comment by rignaneseleo** [comment]
up

mine takes by avg 2s to perform this:

```
@override
  Future<bool> init() async {
    LoggerHelper.d('[FirebaseRemoteConfigRepository] Initializing');

    _remoteConfig = FirebaseRemoteConfig.instanceFor(app: await lazyFirebaseApp.initialize());

    await _remoteConfig.setConfigSettings(
      RemoteConfigSettings(
        fetchTimeout: const Duration(minutes: 1),
        minimumFetchInterval: const Duration(minutes: 1),
      ),
    );

    // Download the config map from the server
    final result = await _remoteConfig.fetchAndActivate();
    if (result) {
      _storeAbValuesInAnalytics();
    }

    LoggerHelper.d('[FirebaseRemoteConfigRepository] Initialized with result: $result');

    return result;
  }
```
